bench/wtperf/wtperf.c

```diff
@@ -755,19 +770,168 @@ worker(void *arg)
+				if (opts->random_value)
+					randomize_value(thread,
+					    value_buf,
+					    workload->modify_delta);
+
+				/*
+				 * Distribute the modifications across whole
+				 * document.
+				 *
+				 * The distribution limits are same as MongoDB
+				 * that are imposed for an update operation to
+				 * be considered as modify or not?
+				 *
+				 * up to 10% of total modification size
+				 * 16 number of modifications per record
+				 *
+				 * NOTE: for records of size less than 160,
+				 * there is no distributed modifications.
+				 */
+
+				nmodifications = MAX_NUM_OF_MODIFICATIONS;
+
+				/* 10% of total size of the record */
+				total_modify_size = strlen(value) /
+				    MAX_SIZE_OF_MODIFICATION;
+
+				/*
+				 * Randomize the maximum modification size
+				 * and offset per modify. With randomization
+				 * of modification offset and size, we can
+				 * produce more number of modifications even
+				 * if the same record is modified again.
+				 */
+				rand_val = __wt_random(&thread->rnd);
+				modify_size =
+				    total_modify_size / (size_t)nmodifications;
+				if (modify_size != 0)
+					modify_size =
+					    (size_t)rand_val % modify_size;
+
+				/*
+				 * Offset location difference between
+				 * modifications
+				 */
+				modify_offset =
+				    strlen(value) / (size_t)nmodifications;
+				if (modify_offset != 0)
+					modify_offset =
+					    (size_t)rand_val % modify_offset;
+
+				/*
+				 * Make sure the offset is more than size,
+				 * otherwise modifications don't spread
+				 * properly.
+				 */
+				if (modify_offset < modify_size)
+				    modify_offset = modify_size + 1;
+
+				for (iter = (size_t)nmodifications;
+					 iter > 0;
+					 iter--)
+					strncpy(
+					    &value_buf[(iter * modify_offset) -
+					    modify_offset],
+					    &value_buf[(iter * modify_offset) -
+					    modify_size],
+					    modify_size);
+
+				/*
+				 * Increase the number of modifications, so
+				 * that normal modify operations gets
+				 * succeeded.
+				 */
+				if (!workload->modify_force_update)
+					nmodifications++;
+
+				oldv.data = value;
+				oldv.size = strlen(value);
+
+				newv.data = value_buf;
+				newv.size = strlen(value_buf);
+
+				/*
+				 * Pass the old and new data to find out the
+				 * modify vectors, according to the passed
+				 * input, this function may fail when the
+				 * modifications count reaches the maximum
+				 * number of modifications that are allowed
+				 * for MODIFY operation.
+				 */
+				ret = wiredtiger_calc_modify(session,
+				    &oldv,
+				    &newv,
+				    total_modify_size,
+				    entries,
+				    &nmodifications);
+
+				if (ret == WT_NOTFOUND ||
+					workload->modify_force_update) {
+					cursor->set_value(cursor, value_buf);
+					if ((ret =
+						cursor->update(cursor)) == 0)
+						break;
+				}
+				else {
+					if ((ret =
+						cursor->modify(cursor,
+						entries,
+						nmodifications)) == 0)
+						break;
+				}
```

Comment:
Lots of line continuation style changes needed above.

Reply:
corrected.